### PR TITLE
feat(globalaccelerator): add support for ip addresses and type

### LINKED ISSUE #28051: globalaccelerator: ip addresses attribute from the Accelerator construct
### Describe the feature

It would be helpful to have the public IP addresses associated with the Accelerator (Global Accelerator) available as properties during stack definition.

### Use Case

As some of applications 3rd party and internally built may require firewall or open port setting requiring the public IP address in configuration, this property (IPAddresses) would be useful.  I haven't found another way (Fn:GetAttr) to be able to get this information easily.

### Proposed Solution

The addition of a read only property of the Accelerator object named something similar to IPAddresses that lists the 2 IP Addresses for the Accelerator object

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.106.0

### Environment details (OS name and version, etc.)

Windows 11, Visual Studio 2022 Community Edition, .Net 6

### PR DESCRIPTION
**Pull request by msambol**
Closes #28051, #28209.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/framework-integ/test/aws-globalaccelerator/test/integ.globalaccelerator-ip-address-type.js.snapshot/global-accelerator-ip-address-type.template.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+ "Resources": {
+  "Accelerator8EB0B6B1": {
+   "Type": "AWS::GlobalAccelerator::Accelerator",
+   "Properties": {
+    "Enabled": true,
+    "Name": "acceleratorWithIpAddressType"
```
Comment (msambol) [inline packages/@aws-cdk-testing/framework-integ/test/aws-globalaccelerator/test/integ.globalaccelerator-ip-address-type.js.snapshot/global-accelerator-ip-address-type.template.json:7]: It passes unit tests but I cannot get `IpAddressType` to show up here. 🤔

**Comment by werebear73-tritelph** [comment]
Thank you for the work on this so far.

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 2]
```diff
@@ -21,6 +21,22 @@ export interface IAccelerator extends cdk.IResource {
    * @attribute
    */
   readonly dnsName: string;
+
+  /**
+   * IP addressess associated with the accelerator.
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:26]:
```suggestion
   * IP addresses associated with the accelerator.
```

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 3]
```diff
@@ -92,10 +164,16 @@ export class Accelerator extends cdk.Resource implements IAccelerator {
     const resource = new ga.CfnAccelerator(this, 'Resource', {
       enabled: props.enabled ?? true,
       name,
+      ipAddresses: props.ipAddresses ?? undefined,
+      // ipAddressType is undefined as a default here for backwards compatibility,
+      // but the service selects "IPV4".
+      ipAddressType: props.ipAddressType ?? undefined,
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:170]: I don't think the null coalescing operators are necessary since they are originally union types with undefined.

```suggestion
      ipAddresses: props.ipAddresses,
      ipAddressType: props.ipAddressType,
```

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 4]
```diff
@@ -92,10 +164,16 @@ export class Accelerator extends cdk.Resource implements IAccelerator {
...
     });
 
     this.acceleratorArn = resource.attrAcceleratorArn;
     this.dnsName = resource.attrDnsName;
+    this.ipAddresses = props.ipAddresses ?? undefined;
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:175]: As the use case, shouldn't we expose `Ipv4Addresses` instead of `props.ipAddresses` specified from props? (In addition, we may want to have `Ipv6Addresses` and `DualStackDnsName`, etc.)

> Ipv4Addresses
> The array of IPv4 addresses in the IP address set. An IP address set can have a maximum of two IP addresses.
> 
> Ipv6Addresses
> The array of IPv6 addresses in the IP address set. An IP address set can have a maximum of two IP addresses.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-globalaccelerator-accelerator.html#aws-resource-globalaccelerator-accelerator-return-values

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 5]
```diff
@@ -40,6 +56,22 @@ export interface AcceleratorProps {
    * @default true
    */
   readonly enabled?: boolean;
+
+  /**
+   * IP addressess associated with the accelerator.
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:61]:
```suggestion
   * IP addresses associated with the accelerator.
```

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 6]
```diff
@@ -55,6 +87,32 @@ export interface AcceleratorAttributes {
...
+ */
+export enum IpAddressType {
+  /**
+   * IPV4
+   */
+  IPV4 = 'IPV4',
+  /**
+   * DUAL_STACK
+   */
+  DUAL_STACK = 'DUAL_STACK',
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:115]:
```suggestion
export enum IpAddressType {
  /**
   * IPV4
   */
  IPV4 = 'IPV4',

  /**
   * DUAL_STACK
   */
  DUAL_STACK = 'DUAL_STACK',
```

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 7]
```diff
@@ -68,6 +126,8 @@ export class Accelerator extends cdk.Resource implements IAccelerator {
     class Import extends cdk.Resource implements IAccelerator {
       public readonly acceleratorArn = attrs.acceleratorArn;
       public readonly dnsName = attrs.dnsName;
+      public readonly ipAddresses = attrs.ipAddresses?? undefined;
+      public readonly ipAddressType = attrs.ipAddressType ?? undefined;
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:130]: I don't think the null coalescing operators are necessary since they are originally union types with undefined.

```suggestion
      public readonly ipAddresses = attrs.ipAddresses;
      public readonly ipAddressType = attrs.ipAddressType;
```

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 8]
```diff
@@ -82,6 +142,18 @@ export class Accelerator extends cdk.Resource implements IAccelerator {
    */
   public readonly dnsName: string;
 
+  /**
+   * IP addressess associated with the accelerator.
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:146]:
```suggestion
   * IP addresses associated with the accelerator.
```

File: packages/aws-cdk-lib/awslint.json [hunk 9]
```diff
@@ -885,6 +885,8 @@
     "props-physical-name:aws-cdk-lib.aws_apigateway.ResourceProps",
     "props-physical-name:aws-cdk-lib.aws_appsync.SourceApiAssociationProps",
     "docs-public-apis:aws-cdk-lib.aws_backup.Schedule",
-    "props-default-doc:aws-cdk-lib.aws_appsync.SourceApi.description"
+    "props-default-doc:aws-cdk-lib.aws_appsync.SourceApi.description",
+    "props-default-doc:aws-cdk-lib.aws_globalaccelerator.AcceleratorAttributes.ipAddresses",
+    "props-default-doc:aws-cdk-lib.aws_globalaccelerator.AcceleratorAttributes.ipAddressType"
```
Comment (go-to-k) [inline packages/aws-cdk-lib/awslint.json:890]: Is there any reason not to write default values in the interface without exclude by `awslint.json`? I would be willing to write it.

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 10]
```diff
@@ -82,6 +142,18 @@ export class Accelerator extends cdk.Resource implements IAccelerator {
...
+
+  /**
+   * The IP address type that an accelerator supports.
+   *
+   * For a standard accelerator, the value can be IPV4 or DUAL_STACK.
+  */
+  public readonly ipAddressType?: IpAddressType;
+
   constructor(scope: Construct, id: string, props: AcceleratorProps = {}) {
     super(scope, id);
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:158]: Like the `validateAcceleratorName` method, it would be good to validate whether the number of elements in IpAddresses is not greater than two in the constructor.

> You can specify one or two addresses, separated by a comma.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-globalaccelerator-accelerator.html#aws-resource-globalaccelerator-accelerator-properties

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 11]
```diff
@@ -82,6 +142,18 @@ export class Accelerator extends cdk.Resource implements IAccelerator {
...
+   * IP addressess associated with the accelerator.
+   */
+  public readonly ipAddresses?: string[];
+
+  /**
+   * The IP address type that an accelerator supports.
+   *
+   * For a standard accelerator, the value can be IPV4 or DUAL_STACK.
+  */
+  public readonly ipAddressType?: IpAddressType;
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:155]:
```suggestion
  /**
   * The IP address type that an accelerator supports.
   *
   * For a standard accelerator, the value can be IPV4 or DUAL_STACK.
   */
  public readonly ipAddressType?: IpAddressType;
```

But, Is it necessary to have even the `ipAddressType` as a property?

**Review by go-to-k — CHANGES_REQUESTED** [review]
Thanks for this PR! I commented some points.

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts (hunk 5 above)
Comment (go-to-k) [reply packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:61]: Also we would be good to add an additional description based on the doc.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-globalaccelerator-accelerator.html#aws-resource-globalaccelerator-accelerator-properties

```
Optionally, if you've added your own IP address pool to Global Accelerator (BYOIP), you can choose IP addresses from your own pool to use for the accelerator's static IP addresses when you create an accelerator. You can specify one or two addresses, separated by a comma. Do not include the /32 suffix.

Only one IP address from each of your IP address ranges can be used for each accelerator. If you specify only one IP address from your IP address range, Global Accelerator assigns a second static IP address for the accelerator from the AWS IP address pool.

Note that you can't update IP addresses for an existing accelerator. To change them, you must create a new accelerator with the new addresses.
```

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 12]
```diff
@@ -21,6 +21,22 @@ export interface IAccelerator extends cdk.IResource {
...
+  readonly ipAddresses?: string[];
+
+  /**
+   * The IP address type that the accelerator supports.
+   *
+   * For a standard accelerator, the value can be IPV4 or DUAL_STACK.
+   *
+   * @attribute
+   */
+  readonly ipAddressType?: IpAddressType;
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:39]: Depending on [this comment](https://github.com/aws/aws-cdk/pull/28055/files#r1400194833), it will be removed.

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 13]
```diff
@@ -55,6 +87,32 @@ export interface AcceleratorAttributes {
...
+   * IP addressess associated with the accelerator.
+   */
+  readonly ipAddresses?: string[];
+
+  /**
+   * The IP address type that an accelerator supports.
+   *
+   * For a standard accelerator, the value can be IPV4 or DUAL_STACK.
+   */
+  readonly ipAddressType?: IpAddressType;
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:101]: Depending on [this comment](https://github.com/aws/aws-cdk/pull/28055/files#r1400194833), it will be removed.

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts (hunk 7 above)
Comment (go-to-k) [reply packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:130]: Depending on [this comment](https://github.com/aws/aws-cdk/pull/28055/files#r1400194833), it will be removed.

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 14]
```diff
@@ -82,6 +142,18 @@ export class Accelerator extends cdk.Resource implements IAccelerator {
    */
   public readonly dnsName: string;
 
+  /**
+   * IP addressess associated with the accelerator.
+   */
+  public readonly ipAddresses?: string[];
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:148]: Depending on [this comment](https://github.com/aws/aws-cdk/pull/28055/files#r1400194833), it will be removed.

**Review by go-to-k — CHANGES_REQUESTED** [review]

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 15]
```diff
@@ -21,6 +21,22 @@ export interface IAccelerator extends cdk.IResource {
    * @attribute
    */
   readonly dnsName: string;
+
+  /**
+   * IP addressess associated with the accelerator.
+   *
+   * @attribute
+   */
+  readonly ipAddresses?: string[];
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:30]: Depending on [this comment](https://github.com/aws/aws-cdk/pull/28055/files#r1400194833), it will be removed.

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 16]
```diff
@@ -55,6 +87,32 @@ export interface AcceleratorAttributes {
    * The DNS name of the accelerator
    */
   readonly dnsName: string;
+
+  /**
+   * IP addressess associated with the accelerator.
+   */
+  readonly ipAddresses?: string[];
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:94]: Depending on [this comment](https://github.com/aws/aws-cdk/pull/28055/files#r1400194833), it will be removed.

**Comment by msambol** [comment]
@go-to-k updated per your feedback. Thanks!

**Review by go-to-k — CHANGES_REQUESTED** [review]
Thanks for the update!

I have commented on some adjustments, please take a look.

File: packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts [hunk 17]
```diff
@@ -40,6 +62,33 @@ export interface AcceleratorProps {
    * @default true
    */
   readonly enabled?: boolean;
+
+  /**
+   * IP addresses associated with the accelerator.
+   *
+   * Optionally, if you've added your own IP address pool to Global Accelerator (BYOIP), you can choose IP
+   * addresses from your own pool to use for the accelerator's static IP addresses when you create an accelerator.
+   * You can specify one or two addresses, separated by a comma. Do not include the /32 suffix.
```
Comment (go-to-k) [inline packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:71]: I think the statement `, separated by a comma` does not necessary because it is for CFn docs.

```suggestion
   * You can specify one or two addresses. Do not include the /32 suffix.
```

Comment (msambol) [reply packages/aws-cdk-lib/aws-globalaccelerator/lib/accelerator.ts:71]: I'd like to keep this as-is, as that's the verbiage in the official [documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-globalaccelerator-accelerator.html#aws-resource-globalaccelerator-accelerator-return-values).

**Comment by werebear73-tritelph** [comment]
Is this ready?  I'm not been part of a issue and PR here.  How long does it usually take for one to get deployed into production?

**Comment by go-to-k** [comment]
@werebear73-tritelph 

There will be a maintainer review after this. I don't know when it will be merged, as it is also up to the maintainer.

**Review by go-to-k — APPROVED** [review]

File: packages/aws-cdk-lib/awslint.json [hunk 18]
```diff
@@ -895,6 +895,9 @@
     "from-method:aws-cdk-lib.aws_apigatewayv2.HttpIntegration",
     "from-method:aws-cdk-lib.aws_apigatewayv2.HttpRoute",
     "from-method:aws-cdk-lib.aws_apigatewayv2.WebSocketIntegration",
-    "from-method:aws-cdk-lib.aws_apigatewayv2.WebSocketRoute"
+    "from-method:aws-cdk-lib.aws_apigatewayv2.WebSocketRoute",
+    "props-default-doc:aws-cdk-lib.aws_globalaccelerator.AcceleratorAttributes.dualStackDnsName",
+    "props-default-doc:aws-cdk-lib.aws_globalaccelerator.AcceleratorAttributes.ipv4Addresses",
+    "props-default-doc:aws-cdk-lib.aws_globalaccelerator.AcceleratorAttributes.ipv6Addresses"
```
Comment (sumupitchayan) [inline packages/aws-cdk-lib/awslint.json:901]: @msambol why were these added to awslint.json?

Comment (msambol) [reply packages/aws-cdk-lib/awslint.json:901]: I get a linter error if they're not in this file.

Comment (sumupitchayan) [reply packages/aws-cdk-lib/awslint.json:901]: The linter error happens when there is no`@default` value specified for the props.

Please include:
```
/**
 * @default <value>
 */
```

in the code comment for your props values and remove the changes to awslint.

Comment (msambol) [reply packages/aws-cdk-lib/awslint.json:901]: @sumupitchayan Pinged you on Slack. These are for imports, I think they'd all be `undefined` ?

Comment (msambol) [reply packages/aws-cdk-lib/awslint.json]: Updated.

**Review by sumupitchayan — APPROVED** [review]
